core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Experiment.java

```diff
@@ -12,234 +12,369 @@
+    // If not saved, then compare resolved actual objects. Otherwise
+    // just compare IDs.
+    if (getId() == UNSAVED_ID || them.getId() == UNSAVED_ID) {
+      if (getName() != null && them.getName() != null) {
+        return getName().equals(them.getName());
+      } else {
+        return getAlias().equals(them.getAlias());
+      }
+    } else {
+      return getId() == them.getId();
+    }
+  }
 
-  /**
-   * Returns the pool of this Experiment object.
-   *
-   * @return Pool pool.
-   */
-  @JsonBackReference(value="pool")
-  public Pool<? extends Poolable> getPool();
+  public String getAccession() {
+    return accession;
+  }
 
-  /**
-   * Sets the kits of this Experiment object.
-   *
-   * @param kits kits.
-   */
-  public void setKits(Collection<Kit> kits);
+  public String getAlias() {
+    return alias;
+  }
 
-  /**
-   * Add a Kit to this Experiment object
-   *
-   * @param kit of type Kit
-   */
-  public void addKit(Kit kit);
+  @Override
+  public List<ChangeLog> getChangeLog() {
+    return changeLog;
+  }
 
-  /**
-   * Returns the kits used to construct this Experiment object.
-   *
-   * @return Collection<Kit> kits.
-   */
-  public Collection<Kit> getKits();
+  public String getDescription() {
+    return description;
+  }
 
-  /**
-   * Return kits used within this experiment of a given KitType
-   *
-   * @param kitType of type KitType
-   * @return Collection<Kit>
-   */
-  @JsonIgnore
-  public Collection<Kit> getKitsByKitType(KitType kitType);
+  @Override
+  public long getId() {
+    return experimentId;
+  }
+
+  public Collection<Kit> getKits() {
+    return kits;
+  }
+
+  public Collection<Kit> getKitsByKitType(KitType kitType) {
+    final ArrayList<Kit> ks = new ArrayList<>();
+    for (final Kit k : kits) {
+      if (k.getKitDescriptor().getKitType().equals(kitType)) {
+        ks.add(k);
+      }
+    }
+    Collections.sort(ks);
+    return ks;
+  }
+
+  public User getLastModifier() {
+    return lastModifier;
+  }
+
+  public Library getLibrary() {
+    return library;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  public Platform getPlatform() {
+    return platform;
+  }
+
+  public List<RunPartition> getRunPartitions() {
+    return runPartitions;
+  }
+
+  @Override
+  public SecurityProfile getSecurityProfile() {
+    return securityProfile;
+  }
+
+  public Study getStudy() {
+    return study;
+  }
+
+  public String getTitle() {
+    return title;
+  }
+
+  @Override
+  @CoverageIgnore
+  public int hashCode() {
+    if (getId() != UNSAVED_ID) {
+      return (int) getId();
+    } else {
+      final int PRIME = 37;
+      int hashcode = 1;
+      if (getName() != null) hashcode = PRIME * hashcode + getName().hashCode();
+      if (getAlias() != null) hashcode = 37 * hashcode + getAlias().hashCode();
+      return hashcode;
+    }
+  }
+
+  @Override
+  public void inheritPermissions(SecurableByProfile parent) throws SecurityException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.SecurityException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)